### PR TITLE
feat: add support for ipns name resolve /ipns/<fqdn>

### PR DESCRIPTION
**Pull request by hugomrdias**
fixes: #1918

needs:
* [x] https://github.com/ipfs/interface-js-ipfs-core/pull/458
* [x] https://github.com/ipfs/js-ipfsd-ctl/pull/332

### CODE REVIEW

**Comment by hugomrdias** [comment]
We can't right now I already opened an issue about it, it breaks name pubsub. I want to merge this and then fix that.

**Comment by lidel** [comment]
@hugomrdias ack on not making it recursive by default in this PR, but unless I am missing some context,  `name resolve` with explicit `--recursive=true` and `--recursive=false` should produce the same result as in go-ipfs, but they don't right now (examples in https://github.com/ipfs/js-ipfs/pull/2002#pullrequestreview-229171650)

**Comment by hugomrdias** [comment]
@lidel i can only reproduce this one 

name resolve /ipns/ipfs.io/images/ipfs-logo.svg --recursive=false

go-ipfs 0.4.20: /ipns/website.ipfs.io/images/ipfs-logo.svg
this PR: /ipns/website.ipfs.io 💢


all the others work as go does.

**Comment by alanshaw** [comment]
@hugomrdias does this supercede https://github.com/ipfs/js-ipfs/pull/1970?

**Comment by hugomrdias** [comment]
> @hugomrdias does this supercede #1970?

yes for ipfs name resolve, i will handle ipfs resolve in another PR

File: test/cli/name-pubsub.js [hunk 1]
```diff
@@ -165,53 +162,6 @@ describe('name-pubsub', () => {
...
-            expect(res).to.satisfy(checkAll([cidAdded]))
-
-            return ipfsA(`name resolve ${nodeBId.id}`)
-          })
-          .then((res) => {
-            expect(res).to.exist()
-            expect(res).to.satisfy(checkAll([cidAdded])) // value propagated to node B
-          })
-      })
-    })
```
[the remark this answers is not in the document]
Comment (hugomrdias) [reply test/cli/name-pubsub.js]: core has tests that do mostly the same thing, i discussed this with vasco and we decided to remove this.

**Comment by hugomrdias** [comment]
> Many tests have been removed?

has we already discussed this PR tries to keep all tests in interface-core or core and cli tests only test cli behaviour.

**Comment by alanshaw** [comment]
> as we already discussed this PR tries to keep all tests in interface-core or core and cli tests only test cli behaviour.

I think that’s the right way, but we need to make sure we have the same coverage as we did before (and preferably more)

**Comment by alanshaw** [comment]
Can we please fix this coverage:

https://codecov.io/gh/ipfs/js-ipfs/compare/88f4e264ee14986c2b2eb16263f747631ef58958...0c833c30fd95125820feab1921a918213d3004bf/changes

Screenshot 2019-06-24 at 10 24 01
Screenshot 2019-06-24 at 10 24 40

**Pull request merged**